jenkinsci/tfs-plugin
# Fuzzy TFS/Team Services URL matching

As a follow-up to #110, we've since discovered a way to support TFS/Team Services Git repository URLs with _and_ without `/DefaultCollection` (see ebdee65b7dde940a1ecf3e2c88962ad64bf6ecef for details).

Manual testing
==============

1. Configure three Jenkins jobs with the Git SCM, pointing to three variations of effectively the same Team Services Git repository URL, making sure to add the **Set build completion status in TFS/Team Services** post-build action:
    1. HTTPS, _with_ `/DefaultCollection`.  For example: `https://fabrikam-fiber-inc.visualstudio.com/DefaultCollection/project/_git/repo`
    1. HTTPS, _without_ `/DefaultCollection`.  For example: `https://fabrikam-fiber-inc.visualstudio.com/project/_git/repo`
    1. SSH, _with_ `/DefaultCollection`.  For example: `ssh://fabrikam-fiber-inc@fabrikam-fiber-inc.visualstudio.com/DefaultCollection/project/_git/repo`
    1. SSH, _without_ `/DefaultCollection`.  For example: `ssh://fabrikam-fiber-inc@fabrikam-fiber-inc.visualstudio.com/project/_git/repo`
2. Manually queue each job once to make sure it's able to fetch from Git using the configured credentials, etc.
3. Trigger a "Git pull request merged" event that contains a Git repository remote URL _without_ `/DefaultCollection`.  For example: `https://fabrikam-fiber-inc.visualstudio.com/project/_git/repo`
    1. Notice that all four jobs were queued.
    2. Notice that the Git plugin reports building the same commit for all four jobs (the commit ID came from the event payload).
    3. Notice that the list of statuses for that commit, in Team Services, includes links back to each of the jobs with the completion status.
4. Trigger a "Git pull request merged" event with URLs _containing_ `/DefaultCollection` in the payload.  For example: `https://fabrikam-fiber-inc.visualstudio.com/DefaultCollection/project/_git/repo`
    1. Notice that all four jobs were queued.
    2. Notice that the Git plugin reports building the same commit for all four jobs (the commit ID came from the event payload).
    3. Notice that the list of statuses for that commit, in Team Services, includes links back to each of the jobs with the completion status.

Mission accomplished!
